.github/workflows/sonar-scanner.yml

```diff
@@ -2,7 +2,9 @@ on:
   # Trigger analysis when pushing in master or pull requests, and when creating
   # a pull request.
   push:
-#    branches:
+   branches:
+     -master
+     -develop
   pull_request:
       types: [opened, synchronize, reopened]
```

Comment:
Each PR merged onto develop or master should be a single commit (squash and merge) so I think triggering on each push makes sense for these branches. Then for pull request we also run to run on each push. Is what I setup correct for this.

Reply:
so we only build develop and master upon direct push. pull requests will triggers on each push and there will be two builds when PR are associated to develop or master. This looks good to me if we still feel like more builds are running parallel we can change it accordingly.

Reply:
No the PR request will only run once since they are not develop or master, they are a feature branch. The reason I think PRs where running duplicate is because we had it configured to run on push for every branch including PRs and then a separate trigger for PRs. There will be no duplicates with this.